edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -130,22 +125,18 @@ const CoursePage: ApplicationRoute = () => {
 		});
 
 	// Filter previous assignments
-	const previousAssignments = assignments.filter(
-		(assignment) => assignment.dueDate.seconds * timeInMS.SECOND <= new Date().getTime()
-	).reverse();
+	const previousAssignments = assignments.filter((assignment) => assignment.dueDate.seconds * timeInMS.SECOND <= new Date().getTime()).reverse();
 
 	// Filter upcoming assignments
-	const upcomingAssignments = assignments.filter(
-		(assignment) => assignment.dueDate.seconds * timeInMS.SECOND > new Date().getTime()
-	);
+	const upcomingAssignments = assignments.filter((assignment) => assignment.dueDate.seconds * timeInMS.SECOND > new Date().getTime());
```

Comment:
Same as above, this should be wrapped in a [`useMemo`](https://react.dev/reference/react/useMemo) as it is not something that should execute at each re-render unless one of its dependencies change.

Reply:
Done!